apps/class-solid/src/components/PermutationsList.tsx

```diff
@@ -0,0 +1,255 @@
+import {
+  type ClassConfig,
+  classConfig,
+  classDefaultConfigSchema,
+} from "@classmodel/class/config";
+import { For, createSignal } from "solid-js";
+import { Button } from "~/components/ui/button";
+import { inflate } from "~/lib/inflate";
+import {
+  type Experiment,
+  type Permutation,
+  deletePermutationFromExperiment,
+  promotePermutationToExperiment,
+  setPermutationConfigInExperiment,
+} from "~/lib/store";
+import { MyTextField, ObjectField } from "./ObjectField";
+import {
+  MdiCakeVariantOutline,
+  MdiCog,
+  MdiDelete,
+  MdiLightVectorDifference,
+} from "./icons";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "./ui/dialog";
+
+function PermutationConfigForm(props: {
+  id: string;
+  onSubmit: (name: string, config: Partial<ClassConfig>) => void;
+  permutationName?: string;
+  config: Partial<ClassConfig>;
+}) {
+  // TODO use reference config as defaults of the schema
+  const schema = classDefaultConfigSchema.definitions?.classConfig;
+  return (
+    <form
+      id={props.id}
+      onSubmit={(event) => {
+        event.preventDefault();
+        const formData = new FormData(event.currentTarget);
+        const rawData = Object.fromEntries(formData.entries());
+        const { name, ...rawDataWithoutName } = rawData;
+        const nameAsString = typeof name === "string" ? name : "";
+        const nestedData = inflate(rawDataWithoutName);
+        // Parse only for validation
+        const data = classConfig.parse(nestedData);
+        // TODO if parse fails, show error
+        props.onSubmit(nameAsString, nestedData);
+      }}
+    >
+      <MyTextField
```

Comment:
My lastpass seems to think this field is a username or password field or so, quite annoying. Not sure why it does that for this field but not for others.

Reply:
It is because the field is called "name". I'll propose a fix below.

Reply:
Its better to turn [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) (add autocompletr="off" attribute) then renaming the name attribue.

Reply:
autocomplete is not enought, so we will also rename.